packages/core/src/metrics/MetricsService.ts

```diff
@@ -72,13 +72,13 @@ export class MetricsService {
       data: {}
     };
 
-    metrics.forEach(m => {
-      payload.data[m.identifier] = m.collect();
-    });
+    for (const m of metrics) {
+      payload.data[m.identifier] = await m.collect();
+    }
```

Comment:
Optional: this runs the promises sequentially which shouldn't be a problem, in order to run them in parallel do:
```javascript
const promises = metrics.concat(this.defaultMetrics)
  .map(m => m.collect().then(d => payload.data[m.identifier] = d));

return promises.then(() => this.publisher.publish(payload));
```

Reply:
I think I prefer the way it is, because it looks clearer, but I don't really have a strong opinion here. I remember @wtrocki was reluctant to remap the arrays though.